pkg/controller/cassandracluster/pod_test.go

```diff
@@ -82,16 +83,48 @@ func TestGetLastOrFirstPod(t *testing.T) {
 	podlist = mkList(mkPod(10, true), mkPod(1, false))
 
 	pod, err = GetLastOrFirstPod(podlist, first)
+	if pod.Status.Phase != v1.PodRunning || pod.DeletionTimestamp != nil {
+		err = fmt.Errorf("Pod is not running")
+	}
 	assertPodIsNotRunning(t, err)
 
 	podlist.Items[0].Status = v1.PodStatus{Phase: v1.PodRunning}
 	ts := metav1.Now()
 	podlist.Items[0].DeletionTimestamp = &ts
 
 	pod, err = GetLastOrFirstPod(podlist, first)
+	if pod.Status.Phase != v1.PodRunning || pod.DeletionTimestamp != nil {
+		err = fmt.Errorf("Pod is not running")
+	}
 	assertPodIsNotRunning(t, err)
 }
 
+
+func TestAllPodContainerReady(t *testing.T) {
+
+	mkPod := func(id int, containers ...bool) *v1.Pod {
+		containerStatuses := []v1.ContainerStatus{}
+	var i int = 0
+		for _, c := range containers{
+			containerStatus := v1.ContainerStatus{Name: fmt.Sprintf("ns-nm-%02d-%02d", id, i), Ready: c}
+			containerStatuses = append(containerStatuses, containerStatus)
+		}
+
+		return &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("ns-nm-%02d", id)},
+			Status: v1.PodStatus{Phase: v1.PodRunning, ContainerStatuses:containerStatuses}}
+	}
+
+	assert.Equal(t, true, AllPodContainerReady(mkPod(1, true)))
+	//No container can't be ready
+	assert.Equal(t, false, AllPodContainerReady(mkPod(1, )))
```

Comment:
empty parameter here ?

Reply:
that was pod with no container, i removed the ,